perses/tests/test_topology_proposal.py

```diff
@@ -0,0 +1,440 @@
+        return 'pdbx'
+    if '.pdb' in filename:
+        return 'pdb'
+    for line in file:
+        if line.startswith('data_') or line.startswith('loop_'):
+            file.seek(0)
+            return 'pdbx'
+        if line.startswith('HEADER') or line.startswith('REMARK') or line.startswith('TITLE '):
+            file.seek(0)
+            return 'pdb'
+    # It's certainly not a valid PDBx/mmCIF.  Guess that it's a PDB.
+    file.seek(0)
+    return 'pdb'
+
+def _matchResidue(res, template, bondedToAtom):
+    atoms = list(res.atoms())
+    if len(atoms) != len(template.atoms):
+        return None
+    matches = len(atoms)*[0]
+    hasMatch = len(atoms)*[False]
+
+    renumberAtoms = {}
+    for i in range(len(atoms)):
+        renumberAtoms[atoms[i].index] = i
+    bondedTo = []
+    externalBonds = []
+    for atom in atoms:
+        bonds = [renumberAtoms[x] for x in bondedToAtom[atom.index] if x in renumberAtoms]
+        bondedTo.append(bonds)
+        externalBonds.append(len([x for x in bondedToAtom[atom.index] if x not in renumberAtoms]))
+
+    residueTypeCount = {}
+    for i, atom in enumerate(atoms):
+        key = (atom.element, len(bondedTo[i]), externalBonds[i])
+        if key not in residueTypeCount:
+            residueTypeCount[key] = 1
+        residueTypeCount[key] += 1
+    templateTypeCount = {}
+    for i, atom in enumerate(template.atoms):
+        key = (atom.element, len(atom.bondedTo), atom.externalBonds)
+        if key not in templateTypeCount:
+            templateTypeCount[key] = 1
+        templateTypeCount[key] += 1
+    if residueTypeCount != templateTypeCount:
+        return None
+
+    if _findAtomMatches(atoms, template, bondedTo, externalBonds, matches, hasMatch, 0):
+        return matches
+    return None
+
+
+def _findAtomMatches(atoms, template, bondedTo, externalBonds, matches, hasMatch, position):
+    if position == len(atoms):
+        return True
+    elem = atoms[position].element
+    name = atoms[position].name
+    for i in range(len(atoms)):
+        atom = template.atoms[i]
+        if ((atom.element is not None and atom.element == elem) or (atom.element is None and atom.name == name)) and not hasMatch[i] and len(atom.bondedTo) == len(bondedTo[position]) and atom.externalBonds == externalBonds[position]:
+            # See if the bonds for this identification are consistent
+
+            allBondsMatch = all((bonded > position or matches[bonded] in atom.bondedTo for bonded in bondedTo[position]))
+            if allBondsMatch:
+                # This is a possible match, so trying matching the rest of the residue.
+
+                matches[position] = i
+                hasMatch[i] = True
+                if _findAtomMatches(atoms, template, bondedTo, externalBonds, matches, hasMatch, position+1):
+                    return True
+                hasMatch[i] = False
+    return False
+
+def _match_template_and_res_name(topology, residue_map, forcefield):
```

Comment:
This test is still currently broken -- I was trying to confirm that the new residue has been built correctly by using forcefield to find a matching template and then asserting that they have the same name (i.e. the residue has a match and also it's the correct template) by copying the first lines of ```createSystem()``` in forcefield.py where the template matches are found.  Something about this method isn't working, as it will find the ```signature``` in ```_templateSignatures``` [(Line 147)] (https://github.com/choderalab/perses/pull/64/files#diff-2551428e6650157ac591946d623ca04dR147) but then does not match the atoms ([Line 149](https://github.com/choderalab/perses/pull/64/files#diff-2551428e6650157ac591946d623ca04dR149): ```matches = None```).  There are then no ```generators``` in ```forcefield._templateGenerators``` which is why the loop at [Line 156] (https://github.com/choderalab/perses/pull/64/files#diff-2551428e6650157ac591946d623ca04dR156) does not lead to a ```template``` or ```Exception```.

I don't think this has to do with the topologies being built incorrectly such that a template match cannot be found, because I can successfully run ```ff.createSystem(new_topology)```.

(Note that the excessive assert and prints are just from me trying to pinpoint where the method was going wrong).